pkg/objectstore/migrate.go

```diff
@@ -328,3 +301,13 @@ func syncBucket(ctx context.Context, src *minio.Client, dst *minio.Client, bucke
 
 	return count, nil
 }
+
+func nilLogger() chan<- string {
+	eternalReads := make(chan string)
+	go func() {
+		for {
```

Comment:
this will leak most definitely. can you rework this? i really dont like this channel as a logger pattern fyi.